app/src/main/java/chat/rocket/android/contacts/worker/ContactSyncWorker.kt

```diff
@@ -36,22 +38,29 @@ class ContactSyncWorker(context : Context, params : WorkerParameters)
     override fun doWork(): Result {
         AndroidWorkerInjection.inject(this)
 
-        val currentServer = serverInteractor.get()!!
+        val currentServer = serverInteractor.get()
+        if (currentServer == null){
+            return Result.RETRY
+        }
         val client: RocketChatClient = factory.create(currentServer)
+        if (!(client.state is State.Connected)){
+            return Result.RETRY
+        }
 
         getContactList()
         val dbManager = dbFactory.create(serverInteractor.get()!!)
 
         val strongHashes: List<String> = (contactArrayList.map { contact -> hashString(contact.getDetail()!!) })
         val weakHashes: List<String> = strongHashes.map{ strongHash -> strongHash.substring(3,9) }
+        var retryFlag:Boolean = false
         runBlocking {
-            val apiResult: List<ContactHolder>? = client.queryContacts(weakHashes)
-            if (apiResult != null) {
-                try {
+            try {
```

Comment:
I'm not an expert in these areas, and probably you have a good reason to do it this way, but is 'runBlocking' the best way to do this?  I notice that most of the async calls in upstream do not use it.

Reply:
Since we're doing it in the worker. All of it would be running essentially in a non-blocking manner on a separate thread than the main thread.

Reply:
cool, thanks.